homeassistant/components/nextbus/config_flow.py

```diff
@@ -0,0 +1,271 @@
+import logging
+
+from py_nextbus import NextBusClient
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import (
+    CONF_AGENCY,
+    CONF_AGENCY_NAME,
+    CONF_ROUTE,
+    CONF_ROUTE_NAME,
+    CONF_STOP,
+    CONF_STOP_NAME,
+    DOMAIN,
+)
+from .util import UserConfig, invert_dict
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _agency_schema(agency_names: Iterable[str]) -> vol.Schema:
+    return vol.Schema({vol.Required(CONF_AGENCY_NAME): vol.In(sorted(agency_names))})
+
+
+def _route_schema(route_names: Iterable[str]) -> vol.Schema:
+    return vol.Schema({vol.Required(CONF_ROUTE_NAME): vol.In(sorted(route_names))})
+
+
+def _stop_schema(stop_names: Iterable[str]) -> vol.Schema:
+    return vol.Schema({vol.Required(CONF_STOP_NAME): vol.In(sorted(stop_names))})
+
+
+def _name_schema(default: str) -> vol.Schema:
+    return vol.Schema({vol.Optional(CONF_NAME, default=default): str})
+
+
+class NextBusFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle Nextbus configuration."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize NextBus config flow."""
+        self.nextbus_config: dict[str, str] = {}
+        self._client = NextBusClient(output_format="json")
+        self._agency_tags: dict[str, str] = {}
+        self._route_tags: dict[str, str] = {}
+        self._stop_tags: dict[str, str] = {}
+        _LOGGER.info("Init new config flow")
+
+    async def _update_agency_config_options(self) -> None:
+        def job():
+            self._agency_tags = {
+                a["title"]: a["tag"] for a in self._client.get_agency_list()["agency"]
+            }
+
+        await self.hass.async_add_executor_job(job)
+
+    async def _update_route_config_options(self, agency_tag: str) -> None:
+        def job():
+            self._route_tags = {
+                a["title"]: a["tag"]
+                for a in self._client.get_route_list(agency_tag)["route"]
+            }
+
+        await self.hass.async_add_executor_job(job)
+
+    async def _update_stop_config_options(
+        self, agency_tag: str, route_tag: str
+    ) -> None:
+        def job():
+            self._stop_tags = {
+                a["title"]: a["tag"]
+                for a in self._client.get_route_config(
+                    route_tag,
+                    agency_tag,
+                )[
+                    "route"
+                ]["stop"]
+            }
+
+        await self.hass.async_add_executor_job(job)
+
+    async def async_step_import(
+        self, config_input: UserConfig | None = None
+    ) -> FlowResult:
+        """Handle import of config."""
+        if not config_input:
+            return self.async_abort(
+                reason="Import failed due to no data to migrate",
+            )
```

Comment:
When would this be `None`? Surely the `import_data` is never `None`.

Reply:
It shouldn't be, but I added checks for edge cases like this because the method input does account for `None` as well.

Reply:
No it shouldn't.
```suggestion
    async def async_step_import(
        self, config_input: dict[str, Any]
    ) -> FlowResult:
        """Handle import of config."""
```